src/daggy.js

```diff
@@ -131,6 +131,26 @@ const arrToString = (arr) => {
   return str + ')'
 }
 
+const makeConstructor= (fields, proto) => {
+  switch (fields.length) {
+    case 1: return function(a) { return makeValue(fields, proto, arguments) }
```

Comment:
In that case 

```js
if (values.length !== fields.length) {
    throw new TypeError(`Expected ${fields.length} arguments, got ${values.length}`)
}
```

doesn't work, because even though user didn't provide enough arguments, we are passing undefined

e.g.
```js
const Tuple = tagged('Tuple', ['x', 'y'])

Tuple(3) // passes [3, undefined] to makeValue so TypeError isn't thrown
```